README.md

```diff
@@ -302,6 +302,56 @@ The list of objects consist of PR names and refs (either a branch or a sha) for
 1. Resolve with the list of objects consists of PR names and refs
 2. Reject if the input or output is not valid
 
+### getPrInfo
+The parameters required are:
+
+| Parameter        | Type  | Required | Description |
+| :-------------   | :---- | :------- | :-------------|
+| config        | Object | true | Configuration Object |
+| config.scmUri | String | true | The scm uri (ex: `github.com:1234:branchName`) |
+| config.token | String | true | Access token for scm |
+| config.prNum | Integer | true | The PR number used to fetch the PR |
+
+#### Expected Outcome
+The object consists of PR name, sha and ref for the pipeline.
+
+#### Expected Promise Response
+1. Resolve with the object consists of PR name, sha and ref
+2. Reject if the input or output is not valid
+
+### getScmContexts
+No parameters are required.
+
+#### Expected Outcome
+The array of scm context name (e.g. [github.com, my-gitlab])
+
+#### Expected Response
+1. The array of scm context name
+
+### canHandleWebhook
+The parameters required are:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| headers        | Object | The request headers associated with the webhook payload |
+| payload        | Object | The webhook payload received from the SCM service |
+
+#### Expected Outcome
+The received webhook is available or not as boolean.
+
+#### Expected Promise Response
+1. Resolve with the received webhook is available or not.
+2. Reject if the input or output is not valid
+
+### getDisplayName
+No parameters are required.
```

Comment:
Exceptions make it impossible to replace with other modules. I think that it is necessary to share the interface, is there any effect on this?